### ISSUE TITLE
Separable Conv2D allows dilation_rate!=1 and strides !=1 

### ISSUE THREAD

**Issue by pokecheater** [keras-team/keras#15283]
Nothing serious :).

I just noticed in tensorflow 2.4 (sry can not update to 2.7 currently since I am stuck to this version because of other compatibilty issues, so I can't tell if this also affects your newest version) that the layer Separable Conv2D states in it's documentations that if I assign another value for the dilation rate than 1 the strides are not to be allowed to have another value than 1 and vice versa.

dilation_rate: An integer or tuple/list of 2 integers, specifying the dilation rate to use for dilated convolution. Currently, specifying any dilation_rate value != 1 is incompatible with specifying any strides value != 1.

https://keras.io/api/layers/convolution_layers/separable_convolution2d/

But in fact it is possible to set both parameters with values > 1.
(Since this is also true (and you will get an exception if you try to apply other values than 1 for both parameters) for the normal covolutional layers I think it is either a bug or a documentation copycat issue.)

Cheers :)

(I have posted this issue already in the tensorflow github repo, they state that I should psot it right here.)

(https://github.com/tensorflow/tensorflow/issues/51710)

**Comment by pokecheater** [comment]
Update: I tested it on a second computer with tensorflow 2.7.0. And was able to see the same behavior. So this seems to be still an issue. Then another question popped up into my mind: 

Why is it forbidden to use dilation rate >1 and stride >1 in conjunction on a normal convolutional layer. In pytorch this seems not to be a problem at all. I also checked some tutorials (again!) how convolutional layers are working. For me there seems no reason while the use of dilation should forbid any striding. In fact a dilation could also be understood as a bigger kernel (where only some points of the kernel will be used), then the stride would just applied like on a normal kernel.

Another question:
From my understanding the convolutional layers in keras or pytorch are based on the same idea/procedure. Since I have not found any documentation in keras on how the output shape is calculated (maybe this is my fault, if there is any documentation on this plz let me know :)), I used the formular from https://pytorch.org/docs/master/generated/torch.nn.Conv2d.html (shape section). But as soon as I use dilation and stride together (in separable_convolution) the output shape is always different from that formular (where it is correct if I leave one of this values at 1).

Thx in advance and sry for so many questions at once. <3
